src/System.Diagnostics.Debug/tests/DebugTests.cs

```diff
@@ -22,14 +25,14 @@ public void Asserts()
             VerifyAssert(() => { Debug.Assert(false, "assert passed", "nothing is wrong {0} {1}", 'a', 'b'); }, "assert passed", "nothing is wrong a b");      
         }
 
-        [Fact]
+        [ConditionalFact(nameof(IsNotDesktopJob))]
```

Comment:
>))] [](start = 47, length = 3)

Where you conditionalize on desktop, can you add a comment expalining what the behavior difference is, for the benefit of future devs? What changed in Debug.Fail?